### PR TITLE
implement a simple perceptron classifier

### PR DESCRIPTION
**Pull request by ajschumacher**
This adds `.perceptron()` for making a single-layer perceptron classifier with `.train()` and `.predict()` methods. Tests are in `perceptron.test.js`.

I haven't added anything to the README etc., but if this looks good that could also be done.

### CODE REVIEW

File: src/simple_statistics.js [hunk 1]
```diff
@@ -254,6 +254,83 @@
...
+    // # [Perceptron Classifier](http://en.wikipedia.org/wiki/Perceptron)
+    //
+    // This is a single-layer perceptron classifier that takes
+    // arrays of numbers and predicts whether they should be classified
+    // as either 0 or 1 (negative or positive examples).
+    function perceptron() {
+        var perceptron_model = {},
+            // The weights, or coefficients of the model;
+            // weights are only populated when training with data.
+            w = [],
```
Comment (tmcw) [inline src/simple_statistics.js]: could this be called `weights` instead of `w`?

File: src/simple_statistics.js [hunk 2]
```diff
@@ -254,6 +254,83 @@
...
+                for (var i = 0; i < w.length; i++) {
+                    w[i] += gradient * features[i];
+                }
+                b += gradient;
+            }
+            return perceptron_model;
+        };
+
+        // Conveniently access the weight array.
+        perceptron_model.w = function() {
```
Comment (tmcw) [inline src/simple_statistics.js]: similarly, should this be `weights`?

File: src/simple_statistics.js (hunk 1 above)
Comment (ajschumacher) [inline src/simple_statistics.js]: Sure could! I'll also change `b` to `bias`.

**Comment by ajschumacher** [comment]
@tmcw Updated! :)

**Comment by tmcw** [comment]
Awesome, merging!

**Pull request merged**